### PR TITLE
Feature/billing

### PR DESCRIPTION
**Pull request by ygrishajev**

### CODE REVIEW

File: apps/api/src/billing/services/master-signing-client/master-signing-client.service.ts [hunk 1]
```diff
@@ -119,6 +123,15 @@ export class MasterSigningClientService {
       const hashes = [...responses.map(hash => toHex(hash.hash)), lastDelivery.transactionHash];
 
       return await Promise.all(hashes.map(hash => client.getTx(hash)));
+    } catch (error) {
+      if (error.message.includes("account sequence mismatch")) {
```
Comment (baktun14) [inline apps/api/src/billing/services/master-signing-client/master-signing-client.service.ts:91]: Does this happen a lot?

File: apps/deploy-web/src/context/WalletProvider/WalletProvider.tsx [hunk 2]
```diff
@@ -175,40 +182,46 @@ export const WalletProvider = ({ children }) => {
...
-        setIsBroadcastingTx(false);
+        setLoadingState(undefined);
       }
 
       if (txResult.code !== 0) {
         throw new Error(txResult.rawLog);
       }
 
-      showTransactionSnackbar("Transaction success!", "", txResult.transactionHash, "success");
+      if (!managedWallet?.isWalletConnected) {
```
Comment (baktun14) [inline apps/deploy-web/src/context/WalletProvider/WalletProvider.tsx]: This feels like a weird condition. If managedWallet is truthy, there will never be a wallet connected. Looking at the hook, we set it to if the managed wallet is thruthy in a ternary for the wallet value. Perhaps we could just check if the userWallet is defined or has an address.

**Comment by baktun14** [comment]
There's currently no way to exit a trial and go back to self custody wallet. We could add the same (...) button with just one option to opt out of the trial and a confirmation?
image

File: apps/api/src/billing/services/master-signing-client/master-signing-client.service.ts (hunk 1 above)
Comment (ygrishajev) [reply apps/api/src/billing/services/master-signing-client/master-signing-client.service.ts:91]: it happened once to me

**Comment by ygrishajev** [comment]
> There's currently no way to exit a trial and go back to self custody wallet. We could add the same (...) button with just one option to opt out of the trial and a confirmation? image

looks like a product decision generally. Let's discuss this with @anilmurty
in the meanwhile I don't really see any reason to add extra into this PR, the feature is incomplete anyway

**Comment by anilmurty** [comment]
Funn

> > There's currently no way to exit a trial and go back to self custody wallet. We could add the same (...) button with just one option to opt out of the trial and a confirmation? image
> 
> looks like a product decision generally. Let's discuss this with @anilmurty in the meanwhile I don't really see any reason to add extra into this PR, the feature is incomplete anyway

nice coincidence. I ran into that just this morning - was trying to show someone a demo of the "Start Trial" button and couldn't get back into that state after starting a trial. let's discuss for sure

**Review by baktun14 — APPROVED** [review]

**Pull request merged**